core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -77,17 +106,50 @@ class SubmissionMonitorActor(val workspaceName: WorkspaceName,
       logger.debug(s"done checking status for submission $submissionId, terminateActor = $terminateActor")
       if (terminateActor) stop(self)
       else scheduleNextMonitorPass
+    case CheckCurrentWorkflowStatusCounts =>
+      logger.debug(s"check current workflow status counts for submission $submissionId")
+      checkCurrentWorkflowStatusCounts pipeTo self
+    case SaveCurrentWorkflowStatusCounts(statusCounts) =>
```

Comment:
Good question! Because `checkCurrentWorkflowStatusCounts` is running in a `Future` (using a thread from the `SlickDataSource` thread pool) and it's only safe to update actor state from the actor thread itself. So I pipe the database Future back to the actor and update the actor state upon receiving a `SaveCurrentWorkflowStatusCounts` message.